pkg/middleware/middleware.go

```diff
@@ -0,0 +1,26 @@
+package middleware
```

Comment:
Added the concept of middleware. These are meant to be used before running a command (cobra command `PreRunE` field) or after running it (cobra command `AfterRunE` field).

These can be used for checking if a user is logged in before executing any other command.

Right now they are used for refreshing the authentication token on the already selected Giant Swarm kubectl context.